compiler/src/middle_end/linearize.re

```diff
@@ -63,16 +63,42 @@ let lookup_symbol = (mod_, mod_decl, name, original_name) => {
   };
 };
 
-type anf_bind =
-  | BSeq(comp_expression)
-  | BLet(Ident.t, comp_expression)
-  | BLetRec(list((Ident.t, comp_expression)))
-  | BLetExport(rec_flag, list((Ident.t, comp_expression)));
-
 type either('a, 'b) =
   | Left('a)
   | Right('b);
 
+let convert_binds = anf_binds => {
+  let void_comp = Comp.imm(Imm.const(Const_void));
+  let void = AExp.comp(void_comp);
+  let (last_bind, top_binds) =
+    switch (anf_binds) {
+    | [bind, ...rest] => (bind, rest)
+    | _ => (BSeq(void_comp), [])
+    };
+  let ans =
+    switch (last_bind) {
+    | BSeq(exp) => AExp.comp(exp)
+    | [@implicit_arity] BLet(name, exp) =>
+      AExp.let_(Nonrecursive, [(name, exp)], void)
```

Comment:
```suggestion
    | BLet(name, exp) => AExp.let_(Nonrecursive, [(name, exp)], void)
```

Reply:
Reason doesn't need implicit_arity afaik. I know we haven't removed all of them from the codebase, but I think we should avoid it in new code.

Reply:
Yeah, it just feels weird to have code that doesn't match the surrounding code.